snuba/datasets/transactions_processor.py

```diff
@@ -231,12 +231,17 @@ def _process_contexts_and_user(
             if context in contexts:
                 del contexts[context]
 
-        transaction_ctx = contexts.get("trace", {})
         # We store trace_id and span_id as promoted columns and on the query level
         # we make sure that all queries on contexts[trace.trace_id/span_id] use those promoted
         # columns instead. So we don't need to store them in the contexts array as well
-        transaction_ctx.pop("trace_id", None)
-        transaction_ctx.pop("span_id", None)
+        #
+        # When we remove the trace_id/span_id from transaction_ctx, it breaks the consumer if the
```

Comment:
We don't want to leave these fields in the context either, they take up an enormous amount of space that we don't need. I think you need to come up with a different solution here.